### ISSUE TITLE
A question

### ISSUE THREAD

**Issue by HIT-leaderone** [LinglanZhao/BiDistFSCIL#2]
Hi, you do fine-tuning for each session's data in test_inc_loop function (even if the call to test_inc_loop in train.py was made at training session 0), won't this violate the definition of CIL? I apologize if my understanding is incorrect.

**Comment by LinglanZhao** [comment]
Thank you for your interest.
Please note that the test_inc_loop function does fine-tuning only if the args **needs_finetune=True** in the nn.Module class [CosClassifier](https://github.com/LinglanZhao/BiDistFSCIL/blob/main/methods/cosine_classifier.py#L27).
As shown in the scripts in the README file, the needs_finetune=False by default in the pretraining phase (train.py), and needs_finetune=True in the testing phase (test.py).
As a result, for model pretraining in session 0, the model is only trained on the base class data when using train.py.
For model selection in session 0, only [the Acc. on base class data](https://github.com/LinglanZhao/BiDistFSCIL/blob/main/train.py#L135) is used, while Acc. on session >= 1 is only used for debug purpose (You can set eval_base_only=True in test_inc_loop to get the same results in train.py).
To summarize, our training and testing pipelines strictly follow the paradigm of CIL.

**Comment by HIT-leaderone** [comment]
Thank you for your reply, I will continue to research&nbsp; on your code.Your work makes a lot of sense and the code is very modular (for replacing encoder with 3D encoder, your code is the most convenient).Thank you a lot!




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "LinglanZhao/BiDistFSCIL"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年7月24日(星期一) 上午10:48
***@***.***&gt;;
抄送:&nbsp;"leader ***@***.***&gt;;"State ***@***.***&gt;;
主题:&nbsp;Re: [LinglanZhao/BiDistFSCIL] A question (Issue #2)





 
Thank you for your interest.
 Please note that the test_inc_loop function does fine-tuning only if the args needs_finetune=True in the nn.Module class CosClassifier.
 As shown in the scripts in the README file, the needs_finetune=False by default in the pretraining phase (train.py), and needs_finetune=True in the testing phase (test.py).
 As a result, for model pretraining in session 0, the model is only trained on the base class data when using train.py.
 For model selection in session 0, only the Acc. on base class data is used, while Acc. on session &gt;= 1 is only used for debug purpose (You can set eval_base_only=True in test_inc_loop to get the same results in train.py).
 To summarize, our training and testing pipelines strictly follow the paradigm of CIL.

**Comment by HIT-leaderone** [comment]
Thank you for your reply, I will continue to research  on your code.Your work makes a lot of sense and the code is very modular (for replacing encoder with 3D encoder, your code is the most convenient).Thank you a lot!
